### DISCUSSION TITLE
Since version 1.2.175, there is a bug in swc's legacyDecorator

### DISCUSSION THREAD

**Discussion by mike199808** [swc-project/swc#6943] [Q&A] [answered]
In version 1.2.174, if I set transform.legacyDecorator, the decorator input parameter has the descriptor & descriptor.initializer attribute, which behaves the same as the babel configuration plugin ['@babel/plugin-proposal-decorators', { 'legacy': true }].

version 1.2.174 link
https://play.swc.rs/?version=1.2.174&code=H4sIAAAAAAAAA22O0QqDMAxF3%2FsVebMFqWyv4hj%2BSVczkXXNaDLGEP%2Fd6jaQYd5yzuUmVQV36qywuj6jl4EidOgpOUEtLvUoJdzwXWbKPg0PoWRGBXk8RaaANlCvN7JWk1I%2BOGZo4ZM8%2FxrXzUEDhx1%2Byfy4wwVZstIGmtO38P98sWSQi3x8UdP6wsZHfEGrjXWmngGwqMw18AAAAA%3D%3D&config=H4sIAAAAAAAAAzXLSwqAMAxF0b28cVfQsRsJMYpiPyQRLKV7l4LO7uDcjtMYsaOSmugsa9npQYRwImM9qiNgFS5KXtQQXW8ZAa6UbSua5nXJTtyWX31ojBcy%2BVj2YQAAAA%3D%3D
However, since version 1.2.175, even if I set transform.legacyDecorator, the decorator input parameter descriptor becomes undefined

version 1.2.175 link
https://play.swc.rs/?version=1.2.175&code=H4sIAAAAAAAAA22O0QqDMAxF3%2FsVebMFqWyv4hj%2BSVczkXXNaDLGEP%2Fd6jaQYd5yzuUmVQV36qywuj6jl4EidOgpOUEtLvUoJdzwXWbKPg0PoWRGBXk8RaaANlCvN7JWk1I%2BOGZo4ZM8%2FxrXzUEDhx1%2Byfy4wwVZstIGmtO38P98sWSQi3x8UdP6wsZHfEGrjXWmngGwqMw18AAAAA%3D%3D&config=H4sIAAAAAAAAAzXLSwqAMAxF0b28cVfQsRsJMYpiPyQRLKV7l4LO7uDcjtMYsaOSmugsa9npQYRwImM9qiNgFS5KXtQQXW8ZAa6UbSua5nXJTtyWX31ojBcy%2BVj2YQAAAA%3D%3D

expectation
I hope that versions after 1.2.175 also get the descriptor & descriptor.initializer attribute in the input parameter of the decorator function, what should I do?

**Comment by mike199808** [comment]
@kdy1  Thank you very much if you could take some time out of your busy schedule to look at this question.

**Reply by mike199808** [reply]
since version 1.2.175, swc's legacyDecorator is different from babel's legacyDecorator

**Reply by mike199808** [reply, marked as answer]
https://github.com/swc-project/swc/issues/7291 look this, author have no plan to support it.

**Comment by shrinktofit** [comment]
Our project also relies on the initializer passed to decorator. @kdy1 Could you please consider providing the old behaviour as an option? I know the decorator stuff is complicated, there're three models:

- TypeScript's one: no initializer, class fields are put in `[[Set]]` semantic.

- Babel's one: initializer is passed to decorator, if a class fields is decorated, it's mandated to have `[[Define]]` semantic.

- Spec one: .... oh it's very different from above.

However, the spec one has not reached to stage4 yet, so actually the current echo still use TypeScript or Babel model.

**Reply by kdy1** [reply]
@shrinktofit It's still supported.

**Reply by kdy1** [reply]
Ah you mean babel version

**Reply by kdy1** [reply]
No. I'll accept PR if it uses the old code as-is, but I'm never going to add it by myself.

**Comment by kdy1** [comment]
Closing as this is not a bug
